filters/auth/grantconfig.go

```diff
@@ -40,27 +40,13 @@ type OAuthConfig struct {
 	// authorization code should be redirected to.
 	CallbackPath string
 
-	// ClientID, the OAuth2 client id of the current service, used to exchange
-	// the access code. Must be set if ClientIDFile is not provided.
-	ClientID string
+	// GetClientID, returns the OAuth2 client id of the current service, used to exchange
+	// the access code
+	GetClientID func() string
```

Comment:
`secrets` package does not provide a suitable `getter` interface.
`SecretsReader` (including `StaticSecret` implementation) require input argument which would be fake or empty string: https://github.com/zalando/skipper/blob/5801bbb44c13d3018490f4cfd1d9e3a50d897913/secrets/readers.go#L6-L14
`SecretSource` returns two-dimensional array: 
https://github.com/zalando/skipper/blob/5801bbb44c13d3018490f4cfd1d9e3a50d897913/secrets/encrypter.go#L19-L21
so I've decided to simply use generic getter function here.

Reply:
I've decided to revert refactoring as it did not improve much